File: src/main/java/io/github/oleksivio/telegram/bot/core/controller/processor/session/ChatSession.java
package io.github.oleksivio.telegram.bot.core.controller.processor.session;

import io.github.oleksivio.telegram.bot.api.model.objects.Update;
import io.github.oleksivio.telegram.bot.api.model.objects.std.Chat;
import io.github.oleksivio.telegram.bot.core.controller.handler.Handler;
import io.github.oleksivio.telegram.bot.core.model.objects.IChat;

import java.util.Objects;
import java.util.Optional;
import java.util.stream.Stream;

public class ChatSession extends SessionManager {

    public ChatSession(OrderManager orderManager,
                       Handler initialHandler,
                       Handler cancelHandler) {
        super(orderManager, initialHandler, cancelHandler);
    }

    @Override
    protected Optional<Long> getIdOptional(Update update) {
        return Stream.of(update.getMessage(),
                         update.getEditedMessage(),
                         update.getChannelPost(),
                         update.getEditedChannelPost())
                .filter(Objects::nonNull)
                .map(IChat::getChat)
                .map(Chat::getId)
                .findFirst();

    }
}

File: src/main/kotlin/io/github/oleksivio/telegram/bot/core/controller/processor/session/ChatSession.kt
package io.github.oleksivio.telegram.bot.core.controller.processor.session

import io.github.oleksivio.tl.kbot.server.api.objects.Update
import io.github.oleksivio.telegram.bot.core.controller.handler.Handler

class ChatSession(orderManager: OrderManager,
                  initialHandler: Handler<*>,
                  cancelHandler: Handler<*>) : SessionManager(orderManager, initialHandler, cancelHandler) {

    override fun getIdOptional(update: Update): Long? {
        return listOfNotNull(update.message,
                update.editedMessage,
                update.channelPost,
                update.editedChannelPost)
                .map { it.chat }
                .map { it.id }
                .firstOrNull()

    }
}
